### PR TITLE
GODRIVER-3123 QE Range Protocol V2

### PR DESCRIPTION
**Pull request by qingyang-hu**
GODRIVER-3123
GODRIVER-3279
GODRIVER-3195

## Summary
- Add an optional parameter `RangeOpts.trimFactor`.
- Make `RangeOpts.sparsity` optional.
- Modify `rangePreview` to `range`.
- Implement prose test "[Range Explicit Encryption applies defaults](https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/README.md#22-range-explicit-encryption-applies-defaults)".
- Sync spec test.
- Update libmongocrypt to 1.11.0
- Document "range" as stable.

### CODE REVIEW

**Review by blink1073 — APPROVED** [review]
LGTM!

File: mongo/options/encryptoptions.go [hunk 1]
```diff
@@ -81,7 +82,7 @@ func (e *EncryptOptions) SetContentionFactor(contentionFactor int64) *EncryptOpt
 	return e
 }
 
-// SetRangeOptions specifies the options to use for explicit encryption with range. It is only valid to set if algorithm is "rangePreview".
+// SetRangeOptions specifies the options to use for explicit encryption with range. It is only valid to set if algorithm is "range".
```
Comment (matthewdale) [inline mongo/options/encryptoptions.go:85]: Should the algorithm be `"Range"`, with a capital "R"?

Comment (matthewdale) [inline mongo/options/encryptoptions.go:85]: Optional: Add a blank line here to make the "Beta" warning more visible.
```suggestion
// SetRangeOptions specifies the options to use for explicit encryption with range. It is only valid to set if algorithm is "range".
//
```

File: mongo/options/encryptoptions.go
Comment (matthewdale) [inline mongo/options/encryptoptions.go:1]: Should we add `Range` to the list of supported values for `EncryptOptions.SetAlgorithm`?

File: mongo/options/encryptoptions.go [hunk 2]
```diff
@@ -105,7 +106,14 @@ func (ro *RangeOptions) SetMax(max bson.RawValue) *RangeOptions {
 // SetSparsity sets the range index sparsity.
 // Beta: The Range algorithm is experimental only. It is not intended for public use. It is subject to breaking changes.
 func (ro *RangeOptions) SetSparsity(sparsity int64) *RangeOptions {
-	ro.Sparsity = sparsity
+	ro.Sparsity = &sparsity
+	return ro
+}
+
+// SetTrimFactor sets the range index trim factor.
```
Comment (matthewdale) [inline mongo/options/encryptoptions.go:113]: Optional: Add a blank line here to make the "Beta" warning more visible.

```suggestion
// SetTrimFactor sets the range index trim factor.
//
```

**Review by blink1073 — APPROVED** [review]

**Review by matthewdale — APPROVED** [review]
Looks good! 👍

**Pull request merged**